### PR TITLE
extension: allow use of ES2018 features

### LINKED ISSUE #5152: Fix browserify to support Object spread/rest across all files
Uncovered that you can do object spread in some of our files but not all. In the bad files, compiling `gulp browserify-lighthouse` in `lighthouse-extension/` fails with `Unexpected token`. Unsure of what triggers the error in some files but not others.

To repro, just put something like `const whatever = {a: 5, ...passContext};` in the `afterPass` of one of the problematic gatherers (like `optimized-images.js`). 

https://github.com/GoogleChrome/lighthouse/pull/5129#issuecomment-386842336 for context:

> Yet another browserify syntax error. It's on object spread again, this time in [lexical-scope](https://www.npmjs.com/package/lexical-scope), called by [insert-module-globals](https://www.npmjs.com/package/insert-module-globals).

> It appears it might be an actual bug, though, and not another parser version that needs to be piped through because you can actually use object spread in some files and not others. e.g. not in `optimized-images` or `accessibility` gatherers, but it's fine in the `websql` or `css-usage` gatherers. I haven't figured out what the error cases have in common yet or how to fix them.

### PR DESCRIPTION
**Pull request by patrickhulce**
**Related Issues/PRs**
closes #5152

### CODE REVIEW

**Comment by patrickhulce** [comment]
alright @paulirish @brendankenny, bring it on!

come at me bro

**Comment by patrickhulce** [comment]
> Interesting that @paulirish's comment isn't what we actually went with.

heh, paul just bumped us to 2.2.0 instead but seems like that goto-bus-stop's fork is really what we wanted :)

> should we just do that and use that branch (maybe to the exact commit instead of just the branch)? It would let us specify the ecmaVersion (and even version of acorn) as an option at that point.

I have a ever so slight preference for not relying on a particular github user's fork and doing the hack patch ourselves, but if you and @paulirish would prefer that resolutions fix, that WFM 👍

File: lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js [hunk 1]
```diff
@@ -185,8 +185,7 @@ class OptimizedImages extends Gatherer {
         }
 
         /** @type {LH.Artifacts.OptimizedImage} */
-        // @ts-ignore TODO(bckenny): fix browserify/Object.spread. See https://github.com/GoogleChrome/lighthouse/issues/5152
```
Comment (underbyte) [inline lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js]: Don't forget about [line 202 - 203](https://github.com/GoogleChrome/lighthouse/blob/d03a05e405e06c25c7696801bd55b9b89c433969/lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js#L202-L203)

Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/dobetterweb/optimized-images.js]: nice catch thanks @underbyte! :)

File: lighthouse-extension/package.json [hunk 2]
```diff
@@ -7,9 +7,11 @@
   "scripts": {
     "watch": "gulp watch",
     "build": "gulp build:production",
+    "debug-build": "node --inspect-brk ./node_modules/.bin/gulp build:production",
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-extension/package.json]: maybe it is, but seeing as the only ndb I have access to is 7 years old https://www.npmjs.com/package/ndb us pleebs have to make due with this 😛

File: lighthouse-extension/gulpfile.js [hunk 3]
```diff
@@ -1,6 +1,18 @@
 // generated on 2016-03-19 using generator-chrome-extension 0.5.4
 
 'use strict';
+
+const fs = require('fs');
+// HACK: patch astw before it's required to use acorn with ES2018
+// see https://github.com/GoogleChrome/lighthouse/issues/5152
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-extension/gulpfile.js]: done

File: lighthouse-extension/gulpfile.js [hunk 4]
```diff
@@ -1,6 +1,18 @@
...
+const acornPath = require.resolve('acorn');
+const astwPath = require.resolve('astw/index.js');
+const astwOriginalContent = fs.readFileSync(astwPath, 'utf8');
+const astwPatchedContent = astwOriginalContent
+  .replace(/ecmaVersion: .* 8,/, 'ecmaVersion: 2018,')
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-extension/gulpfile.js]: done

File: lighthouse-extension/gulpfile.js [hunk 5]
```diff
@@ -1,6 +1,18 @@
...
+  .replace(/require\('acorn'\)/, `require(${JSON.stringify(acornPath)})`);
```
[the remark this answers is not in the document]
Comment (patrickhulce) [reply lighthouse-extension/gulpfile.js]: done

**Pull request merged**